advisors/spring-ai-advisors-vector-store/src/main/java/org/springframework/ai/chat/client/advisor/vectorstore/VectorStoreChatMemoryAdvisor.java

```diff
@@ -87,80 +89,76 @@ public static Builder builder(VectorStore chatMemory) {
 	}
 
 	@Override
-	public AdvisedResponse aroundCall(AdvisedRequest advisedRequest, CallAroundAdvisorChain chain) {
+	public ChatClientResponse adviseCall(ChatClientRequest chatClientRequest, CallAdvisorChain callAdvisorChain) {
```

Comment:
`PgVectorStoreWithChatMemoryAdvisorIT` now fails with 

```
java.lang.AssertionError: 
Expecting actual:
  UserMessage{content='joke', properties={messageType=USER}, messageType=USER}
to be an instance of:
  org.springframework.ai.chat.messages.SystemMessage
but was instance of:
  org.springframework.ai.chat.messages.UserMessage
  ```

Reply:
System messages shouldn't be added to the vector store for a similarity search I guess but generally speaking having the system prompt as part of the conversation is needed.

Reply:
There was some inconsistency in behaviour between the two cases where we try to augment a SystemMessage that doesn't exist and one that does.

I fixed that and also extended the `PgVectorStoreWithChatMemoryAdvisorIT` to validate both scenarios.

Reply:
thanks!